src/journal/test-journal-append.c

```diff
@@ -0,0 +1,273 @@
+        uint64_t iteration_step = 1;
+        uint64_t corrupt_step = 31;
+        bool sequential = false;
+        bool run_one = false;
+        int c, r;
+
+        test_setup_logging(LOG_DEBUG);
+
+        enum {
+                ARG_START_OFFSET = 0x1000,
+                ARG_ITERATIONS,
+                ARG_ITERATION_STEP,
+                ARG_CORRUPT_STEP,
+                ARG_SEQUENTIAL,
+                ARG_RUN_ONE,
+        };
+
+        static const struct option options[] = {
+                { "help",                no_argument,       NULL, 'h'                     },
+                { "start-offset",        required_argument, NULL, ARG_START_OFFSET        },
+                { "iterations",          required_argument, NULL, ARG_ITERATIONS          },
+                { "iteration-step",      required_argument, NULL, ARG_ITERATION_STEP      },
+                { "corrupt-step",        required_argument, NULL, ARG_CORRUPT_STEP        },
+                { "sequential",          no_argument,       NULL, ARG_SEQUENTIAL          },
+                { "run-one",             required_argument, NULL, ARG_RUN_ONE             },
+                {}
+        };
+
+        assert_se(argc >= 0);
+        assert_se(argv);
+
+        while ((c = getopt_long(argc, argv, "", options, NULL)) >= 0)
+                switch (c) {
+
+                case 'h':
+                        printf("Syntax:\n"
+                               "  %s [OPTION...]\n"
+                               "Options:\n"
+                               "    --start-offset=OFFSET   Offset at which to start corrupting the journal\n"
+                               "                            (default: random offset is picked, unless\n"
+                               "                            --sequential is used - in that case we use 0 + iteration)\n"
+                               "    --iterations=ITER       Number of iterations to perform before exiting\n"
+                               "                            (default: 100)\n"
+                               "    --iteration-step=STEP   Iteration step (default: 1)\n"
+                               "    --corrupt-step=STEP     Corrupt every n-th byte starting from OFFSET (default: 31)\n"
+                               "    --sequential            Go through offsets sequentially instead of picking\n"
+                               "                            a random one on each iteration. If set, we go through\n"
+                               "                            offsets <0; ITER), or <OFFSET, ITER) if --start-offset=\n"
+                               "                            is set (default: false)\n"
+                               "    --run-one=OFFSET        Single shot mode for reproducing issues. Takes the same\n"
+                               "                            offset as --start-offset= and does only one iteration\n"
+                               , argv[0]);
```

Comment:
`program_invocation_short_name` is better?

Reply:
Ah, interesting, didn't know about this one.